test/javascript/turbolinks_replace_test.coffee

```diff
@@ -376,3 +376,23 @@ suite 'Turbolinks.replace()', ->
       assert.equal @window.count, 1 # using importNode before swapping the nodes would double-eval scripts in Chrome/Safari
       done()
     @Turbolinks.replace(doc, change: ['change'])
+
+  test "doesn't run scripts inside :permanent nodes", (done) ->
+    doc = """
+      <!DOCTYPE html>
+      <html>
+      <head>
+        <title>title</title>
+      </head>
+      <body>
+        <div id="change">
+          new content
+        </div>
+      </body>
+      </html>
+    """
+    @document.addEventListener 'page:partial-load', =>
+      assert.equal @window.countPerm, 1
```

Comment:
I would prefer you add assertions [here](https://github.com/rails/turbolinks/blob/27318d2f0ab37e0abfee405e1bb85dde7d90f226/test/javascript/turbolinks_visit_test.coffee#L56) and [here](https://github.com/rails/turbolinks/blob/27318d2f0ab37e0abfee405e1bb85dde7d90f226/test/javascript/turbolinks_visit_test.coffee#L107) instead of making a new test (also, the behavior isn't specific to partial replacement, since permanent nodes are also replaced on full page visit/replace).

We should also add a test asserting that `data-turbolinks-eval="always"` scripts get run every time.